### DISCUSSION TITLE
Error exporting metrics with ReadStatisticsFromData enabled (retry without statistics): Query (14, 20) Column [Parameter_XXXX] is part of composite key, but not all columns of the composite key are included in the  expression or its dependent expression.

### DISCUSSION THREAD

**Discussion by amien90** [DaxStudio/DaxStudio#1314] [General]
I think i'm not the first person that asked this question, but i can't figure it out.

I get an error when i try to export the VPAX file : Error exporting metrics with ReadStatisticsFromData enabled (retry without statistics): Query (14, 20) Column [Parameter_XXXX] is part of composite key, but not all columns of the composite key are included in the expression or its dependent expression.

Do i first need to fix something? What does this error actually mean? What needs to be fixed?

Help much appraised.

**Comment by marcosqlbi** [comment]
Are you using DirectQuery, DirectLake, or a composite model? Or do you import all the tables?

**Comment by amien90** [comment]
All import storage mode

**Comment by marcosqlbi** [comment]
Can you reproduce the issue on a Power BI file that you can share? I know it could take time, but it could be something related to a specific table and the model you have, so if you remove tables and isolate the one that raises the issue, you may then create a version with dummy data that generates the problem and that we can analyze. Thanks!

**Comment by amien90** [comment]
Hi Marco .. Thanks for helping out .. i will look what is wrong with this table. It is a table island (not connected to the model at all).. i removed the table for now .. and it works

**Comment by marcosqlbi** [comment]
@amien90 it's possible that this is the same issue as this one: https://github.com/TabularEditor/TabularEditor3/issues/1376
We already have a fix for the VertiPaq Analyzer library.
@dgosbell make sure you update VPAX libraries in the next DAX Studio release, thanks!
